test/test_spark.py

```diff
@@ -92,8 +94,8 @@ def test_fn(x):
                 pass
 
     with pytest.raises(Exception):
-        with parallel_backend('spark', n_jobs=4):
-            Parallel()(delayed(test_fn)(i) for i in range(4))
+        with parallel_backend('spark', n_jobs=2):
+            Parallel()(delayed(test_fn)(i) for i in range(2))
```

Comment:
Not the scope of this PR, we should add `match`to `pytest.raises`:

```python
with pytest.raises(Exception, match="..."):
    ...
```

Reply:
The exception message is different between pyspark 3.2 and lower pyspark versions. We can improve it in followup PR